lib/ui/painting.dart

```diff
@@ -1681,6 +1700,7 @@ class Image {
   /// useful when trying to determine what parts of the program are keeping an
   /// image resident in memory.
   void dispose() {
+    onDispose?.call(this);
```

Comment:
Here and elsewhere: can we guard calling these so it's compiled out of release mode?

Reply:
Discussed offline - any flag we put in here would be at best a runtime check that's probably no faster than the null check here.